pkgs/development/python-modules/pip/default.nix

```diff
@@ -36,14 +43,48 @@ buildPythonPackage rec {
     installShellFiles
     setuptools
     wheel
+
+    # docs
+    sphinx
+    sphinx-copybutton
+    sphinx-inline-tabs
+    sphinxcontrib-towncrier
+    myst-parser
+  ];
+
+  sphinxBuilders = [
+    "man"
+  ];
+
+  outputs = [
+    "out"
+    "man"
   ];
 
+  postBuild = ''
+    cd docs
+    # remove intersphinx extension as it reaches out to the internet and is not needed for man page generation
+    sed -i '/"sphinx.ext.intersphinx",/d' html/conf.py
```

Comment:
Reading [the file](https://github.com/pypa/pip/blob/23.2.1/docs/html/conf.py) now, it should be safe, or even safer to use `substituteInPlace`. This way you'd at least get a warning in the future when the substitution doesn't apply anymore.